packages/core/src/_model-internals/get-belongs-to-associations-with-target.ts

```diff
@@ -0,0 +1,22 @@
+import { BelongsToAssociation } from '../associations/index.js';
+import type { ModelStatic } from '../model.js';
+
+/**
+ * Returns all BelongsTo associations in the entire Sequelize instance that target the given model.
+ *
+ * @param target
+ */
+export function getBelongsToAssociationsWithTarget(target: ModelStatic): BelongsToAssociation[] {
```

Comment:
had to move it to its own file to avoid a circular import

We should consider adding a plugin to ban internal circular imports (if possible for our codebase)

I put it in a `_` prefixed folder as I expect we'll eventually use `sync-exports` for this package too

Reply:
sync-exports is indeed something we want for all packages, including core. The more the packages are similar in structure, the easier it becomes to work on different packages